docs/engineering/android-app/configuring/data-migration.mdx

```diff
@@ -0,0 +1,88 @@
+# Data migration
+
+
+The main functionalities of data migration are:
+- Migrating data by updating resources based on a set of data queries and key-value pairs,
+- Retrieving the latest migration version from shared preferences,
+- Performing the migration only if the new version is greater than or equal to the client's latest stored migration version.
+
+The data migration configuration outlines a set of rules and operations for migrating FHIR resources, specifically `Task` resource migration. With this configuration one is able to filter the target `Task`, read its referenced `PlanDefinition` and `rev_including` its careplans then filter for the task that lies within the careplan period and read the resulting careplan id then use it to update the `task.basedOn`. It ensures that all the resources that are being generated through `StructureMap`s are not malformed.
```

Comment:
Some questions on this:

1. Do migrations only work for `Task`s or is this an example and they can be applied to other resources as well?
2. What does `rev_including` mean?
3. Why does it need to update the `task.basedOn` field?
4. How does it ensure that the resource generated by SMs are not malformed?

Reply:
1. Task should be just an example resource targetted by the migration scripts written
2. In FHIR Search API Forward inclusion with _include adds resources and resource versions referenced by the primary results, and reverse inclusion with _revinclude adds resources that reference the primary results.
3. Again this is just an example resource `attribute` targetted by the migration scripts written it can be any attribute within the resource, only noting that currently, we can only update an already existing attribute and not create a new attribute
4. This is not true, @sharon2719 lets remove the statement or reword it to` it cleans out the malformed resource attributes that a wrong StructureMap may have generated`